Makefile

```diff
@@ -90,14 +90,12 @@ build-wasm:
 
 .PHONY: format
 format:
-	$(black)
-	$(ruff) --fix --exit-zero
+	ruff format --check --diff $(sources) 
 	cargo fmt
 
 .PHONY: lint-python
 lint-python:
-	$(ruff)
-	$(black) --check --diff
+	ruff --fix --exit-zero $(sources) 
```

Comment:
I think we'll want to do something like we did with `pydantic` here:

https://github.com/pydantic/pydantic/blob/9868b456e4352558aa01217a6688b30554c6f290/Makefile#L26-L34

With a `ruff --fix` and `ruff format` call in the `format` step and `ruff` and `ruff format check` in the lint step 👍

Reply:
Correct, lint should change anything.

Surely `ruff format` should imply `ruff --fix`?

Reply:
I think we should use both based on this: https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules

Reply:
@sydney-runkle 

My thinking is there was some redundancy in the last PR, imitated from previous code. 
As an example, a linter and formatter command under the linting step:

```
.PHONY: lint  ## Lint python source files
lint: .pdm
	pdm run ruff $(sources)
	pdm run ruff format --check $(sources)
```

Nonetheless happy to make the modifications per your recommendation.

Reply:
Yeah, I think we're really using lint like a `check` step and the format step like a `fix` step. I'm making some changes to `pytest-examples` right now that should iron out the details of which things we want to use where.

I'll get back to you on Monday with an answer about which we want to use here 👍

Reply:
Let's stick with the same pattern we're using in `pydantic` for now 👍

Reply:
Done, let me know if anything else.